src/d3d12/d3d12_textures.cpp

```diff
@@ -102,21 +111,94 @@ namespace wr::d3d12
 		D3D12_SHADER_RESOURCE_VIEW_DESC srv_desc = {};
 		srv_desc.Shader4ComponentMapping = D3D12_DEFAULT_SHADER_4_COMPONENT_MAPPING;
 		srv_desc.Format = (DXGI_FORMAT)tex->m_format;
-		srv_desc.Texture2D.MipLevels = tex->m_mip_levels;
 
 		//Calculate dimension
 		D3D12_SRV_DIMENSION dimension;
 
 		if (tex->m_is_cubemap)
 		{
 			dimension = D3D12_SRV_DIMENSION_TEXTURECUBE;
+			srv_desc.TextureCube.MipLevels = tex->m_mip_levels;
+		}
+		else
+		{
+			if (tex->m_depth > 1)
+			{
+				//Then it's a 3D texture
+				dimension = D3D12_SRV_DIMENSION_TEXTURE3D;
+				srv_desc.Texture3D.MipLevels = tex->m_mip_levels;
+			}
+			else
```

Comment:
can't this be `else if`?

Reply:
nope, those tests need to be executed only if the if statement fails